### PR TITLE
executor: remove loading unnecessary stats during analyze partition table in dynamic mode 

### PR DESCRIPTION
**Pull request by Yisaer**
Signed-off-by: yisaer <disxiaofei@163.com>

### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/37977

Problem Summary:

### What is changed and how it works?

remove loading unnecessary stats during analyze partition table in dynamic mode

### Check List

Tests 

- [x] Unit test
- [x] Manual test (add detailed scripts or steps below)

https://pingcap.feishu.cn/docx/doxcnQxjfgkAbxm2Z2r3ATsBfSf

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by hawkingrei** [comment]
@Yisaer PTAL
```
statistics/handle/handle.go:462:3: should merge variable declaration with assignment on next line (S1021)
```

File: statistics/handle/handle.go [hunk 1]
```diff
@@ -413,11 +416,59 @@ func (h *Handle) MergePartitionStats2GlobalStatsByTableID(sc sessionctx.Context,
...
+
+// GetTableAllPartitionStats return all partition stats of a table
+func (h *Handle) GetTableAllPartitionStats(is infoschema.InfoSchema, physicalID int64) (map[int64]*statistics.Table, error) {
+	h.mu.Lock()
+	globalTable, ok := h.getTableByPhysicalID(is, physicalID)
+	h.mu.Unlock()
+	if !ok {
+		return nil, errors.Errorf("unknown physical ID %d in stats meta table, maybe it has been dropped", physicalID)
+	}
+	globalTableInfo := globalTable.Meta()
```
Comment (xuyifangreeneyes) [inline statistics/handle/handle.go:430]: Maybe there exists some way to get `TableInfo` faster than `get statistics.Table, then get TableInfo`.

Comment (Yisaer) [reply statistics/handle/handle.go]: removed.

File: statistics/handle/ddl.go [hunk 2]
```diff
@@ -149,7 +149,7 @@ func (h *Handle) updateGlobalStats(tblInfo *model.TableInfo) error {
 		opts[ast.AnalyzeOptNumBuckets] = uint64(globalColStatsBucketNum)
 	}
 	// Generate the new column global-stats
-	newColGlobalStats, err := h.mergePartitionStats2GlobalStats(h.mu.ctx, opts, is, tblInfo, 0, nil)
+	newColGlobalStats, err := h.mergePartitionStats2GlobalStats(h.mu.ctx, opts, is, tblInfo, 0, nil, nil)
```
Comment (xuyifangreeneyes) [inline statistics/handle/ddl.go:152]: Can we also avoid loading partition stats repeatedly in `updateGlobalStats`

**Review by xuyifangreeneyes — APPROVED** [review]
rest LGTM

File: statistics/handle/ddl.go (hunk 2 above)
Comment (Yisaer) [reply statistics/handle/ddl.go:152]: sure, this will be revised in the following pr.

**Review by qw4990 — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**